Page type common bundle experiment

andrewscfc opened the pull request:
**Overall change:**
This builds on https://github.com/bbc/simorgh/tree/code-split-page-type-bundles and delivers a webpack config with newly introduced `common` bundles that are extracted from all the bundles in the app:
![image](https://user-images.githubusercontent.com/9645462/88309760-ad58d780-cd06-11ea-9d59-459ae8e6241c.png)

They appear to be bundles containing common modules found on different page types, if you visit http://localhost:7080/mundo/media-23283126 on this branch you will see only the common bundles that include modules used on MediaAsset pages are downloaded:
![image](https://user-images.githubusercontent.com/9645462/88310331-70d9ab80-cd07-11ea-86cd-3a26efa63620.png)

`common-FrontPage-IdxPage-31ecd969.39680be8` has not been downloaded because none of the common modules in that bundle need to be downloaded.

Still lots of work to do on this but I think it's a starting point to making sure our page weight actually goes down overall, the original PR, decreased the initial page weight for a visit to an individual page but then increased it for subsequent page type visits as there was duplication between each page type bundle, this PR begins to solve that problem: https://github.com/bbc/simorgh/pull/7229#discussion_r458622840 


**Code changes:**

- _A bullet point list of key code changes that have been made._
- _When describing code changes, try to communicate **how** and **why** you implemented something a specific way, not just **what** has changed._

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

karinathomasbbc committed:
Adding a custom command to check that a URL returns the expected status code

karinathomasbbc committed:
Renaming command to testResponseCode

karinathomasbbc committed:
Update assertion message

Commit:
Merge branch 'latest' into update-code-owners

karinathomasbbc committed:
Merge branch 'latest' into server-refactor

karinathomasbbc committed:
Merge branch 'latest' into stabilise-on-demand-e2es

andrewscfc committed:
WIP introducing commons bundles specific to page types

karinathomasbbc committed:
Merge branch 'latest' into liveradio_schedule

joshcoventry committed:
Merge branch 'latest' into 6918-increase-manifest-cache-control

andrewscfc commented on scripts/bundleSize.js:
```diff
@@ -141,13 +146,16 @@ vendorBundleData.forEach(bundle => vendorBundlesTable.push(bundle));
 console.log(mainBundlesTable.toString());
 console.log(vendorBundlesTable.toString());
 console.log(pageBundlesTable.toString());
+console.log(commonBundlesTable.toString());
```
re-added common bundles tooling from your other branch: https://github.com/bbc/simorgh/compare/code-split-page-type-bundles-experimentation

andrewscfc commented on scripts/bundleSizeConfig.js:
```diff
@@ -1,6 +1,6 @@
-  MIN_SIZE: 626,
-  MAX_SIZE: 665,
+  MIN_SIZE: 0,
+  MAX_SIZE: 10000,
```
Made this massive to stop it blocking my testing

andrewscfc commented on src/app/pages/index.js:
```diff
@@ -20,7 +20,7 @@ const applyErrorPageHandlers = pipe(withLoading, withPageWrapper, withContexts);
   applyBasicPageHandlers,
   withVariant,
-)(loadable(() => import(/* webpackPrefetch: true */ './ArticlePage')));
+)(loadable(() => import('./ArticlePage')));
```
removed pre-fetching as it's a distraction initially, can be re-added at any point

andrewscfc commented on src/server/Document/index.jsx:
```diff
@@ -26,7 +26,7 @@ const renderDocument = async ({
   const extractor = new ChunkExtractor({ statsFile });
 
-  const linkTags = extractor.getLinkTags();
+  const linkTags = extractor.getLinkElements();
```
fix as per https://github.com/bbc/simorgh/compare/code-split-page-type-bundles-experimentation

andrewscfc commented on webpack.config.client.js:
```diff
@@ -73,12 +73,18 @@ module.exports = ({
         maxSize: 245760, // 240kb
         cacheGroups: {
+          common: {
+            name: true,
```
This gives a nice readable name to show which bundles the common elements have been extracted from: 
![image](https://user-images.githubusercontent.com/9645462/88309006-ab424900-cd05-11ea-82bc-5a58e692947a.png)
In production they recommend turning this off but it's great for debugging: https://webpack.js.org/plugins/split-chunks-plugin/#splitchunksname

karinathomasbbc committed:
Retrigger builds

andrewscfc commented on webpack.config.client.js:
```diff
@@ -73,12 +73,18 @@ module.exports = ({
         cacheGroups: {
+          common: {
+            name: true,
+            minChunks: 2,
```
if two chunks e.g. MediaAssetPage and PhotoGallery share the same component, this will be split out into a common chunk. If this were three it would need another page type bundle to share a module before it was split out.

andrewscfc commented on webpack.config.client.js:
```diff
@@ -73,12 +73,18 @@ module.exports = ({
+          common: {
+            name: true,
+            minChunks: 2,
+            chunks: 'all',
```
This ensures that `async` chunks are considered for spliting, this is the case for all our new 'loadable' page type bundles.

rhenshaw56 committed:
Merge branch 'latest' into recommendation-skip-link

rhenshaw56 committed:
switched id to kebab case and used psammead-detokeniser

rhenshaw56 committed:
Merge branch 'recommendation-skip-link' of https://github.com/bbc/simorgh into recommendation-skip-link

karinathomasbbc committed:
wip separate toggles into separate environment file

karinathomasbbc committed:
Merge pull request #7212 from bbc/liveradio_schedule

Radio schedule on Live Radio pages

karinathomasbbc committed:
Merge branch 'latest' into stabilise-on-demand-e2es

karinathomasbbc committed:
Merge branch 'latest' into server-refactor

Commit:
Merge branch 'latest' into related-images

karinathomasbbc committed:
Merge pull request #7250 from bbc/stabilise-on-demand-e2es

Stabilise on demand e2es

karinathomasbbc committed:
Merge branch 'latest' into server-refactor

chris-hinds commented:
Nice work @andrewscfc and @jroebu14 does the idea of common bundle mean that all components in the bundle are shared between two page types? or do you get a common bundle if say two page types have the image component? I guess what I am asking is will the common bundle also include components that are not common?

Commit:
Merge branch 'latest' into 6918-increase-manifest-cache-control

rhenshaw56 committed:
Merge branch 'latest' into recommendation-skip-link

RichardPK committed:
Merge branch 'latest' into update-launch-dates-od-tv-branch-2

sareh committed:
Merge pull request #7310 from bbc/6918-increase-manifest-cache-control

Add cache control to the manifest.json response header

RichardPK committed:
Merge branch 'latest' into update-launch-dates-od-tv-branch-2

karinathomasbbc committed:
Merge branch 'latest' into adhoc-cypress-tests

karinathomasbbc committed:
Merge branch 'adhoc-cypress-tests' into adhoc-legacy-media-playback-tests

karinathomasbbc committed:
Adding recommendations logic as it was accidentally removed during merge

karinathomasbbc committed:
Merge branch 'server-refactor' of github.com:bbc/simorgh into server-refactor

rhenshaw56 committed:
Merge branch 'latest' into recommendation-skip-link

karinathomasbbc committed:
Merge branch 'latest' into server-refactor

rhenshaw56 committed:
incoporated review feedback

rhenshaw56 committed:
updated snapshots

rhenshaw56 committed:
Merge branch 'recommendation-skip-link' of https://github.com/bbc/simorgh into recommendation-skip-link

karinathomasbbc committed:
Merge pull request #7303 from bbc/update-launch-dates-od-tv-branch-2

Add release dates for Batch 2 of OD-TV services

karinathomasbbc committed:
Merge branch 'latest' into server-refactor

jimjohnsonrollings committed:
Add 40 pngs

rhenshaw56 committed:
removed skip namespacing

rhenshaw56 committed:
changed textEndId to endTextId

rhenshaw56 committed:
Merge branch 'latest' into recommendation-skip-link

karinathomasbbc committed:
Merge pull request #7279 from bbc/server-refactor

Server Refactor - move local server setup to separate file

paruchurisilpa committed:
Merge branch 'latest' into recommendation-skip-link

jimjohnsonrollings committed:
Make promo assets render in index alsos

jimjohnsonrollings committed:
Update container and tests

jimjohnsonrollings committed:
Update test

karinathomasbbc committed:
Merge branch 'latest' into related-images

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into adhoc-cypress-tests

karinathomasbbc committed:
Set chromeWebSecurity to false

karinathomasbbc committed:
Delete example.json

karinathomasbbc committed:
Updating read me

karinathomasbbc committed:
Merge branch 'adhoc-cypress-tests' of github.com:bbc/simorgh into adhoc-cypress-tests

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into toggles-file-per-environment

karinathomasbbc committed:
Toggles in alphabetical order

karinathomasbbc committed:
Adding a test toggle for testing purposes

karinathomasbbc committed:
Fix toggle tests

karinathomasbbc committed:
Adding snapshot tests for toggles

karinathomasbbc committed:
Use default export

karinathomasbbc committed:
Move environment config for toggles into a separate file (doesn't need to be a directory)

karinathomasbbc committed:
Add whitespace

karinathomasbbc committed:
Update snapshots

karinathomasbbc committed:
Adding descriptions for all toggles currently available for Simorgh application

karinathomasbbc committed:
Adding a const file for all toggle names for reusability

karinathomasbbc committed:
Merge branch 'toggles-file-per-environment' of github.com:bbc/simorgh into toggles-file-per-environment

karinathomasbbc committed:
Update README.md

karinathomasbbc committed:
Update README.md

karinathomasbbc committed:
Update README.md

karinathomasbbc committed:
Update liveConfig.js

karinathomasbbc committed:
Update localConfig.js

karinathomasbbc committed:
Update testConfig.js

karinathomasbbc committed:
Update snapshot

karinathomasbbc committed:
Updating snapshots

karinathomasbbc committed:
Update README.md

karinathomasbbc committed:
Rename const file

karinathomasbbc committed:
Merge branch 'toggles-file-per-environment' of github.com:bbc/simorgh into toggles-file-per-environment

rhenshaw56 committed:
prevent summary rendering in secondary column at medium gel widths

HarveyPeachey committed:
Toggles recommendations to on in test

rhenshaw56 committed:
updated snapshots

HarveyPeachey committed:
Merge pull request #7322 from bbc/enable-recommendations-testenv

Enables STY Recommendations on test

rhenshaw56 committed:
Merge branch 'latest' into recommendation-skip-link

Commit:
Merge branch 'latest' into related-images

ryanmccombe committed:
Merge branch 'latest' into tv-cinema-mode-live

rhenshaw56 committed:
updated more snapshots

Commit:
Merge pull request #7304 from bbc/related-images

Fix blurry Story promo images at higher resolutions

rhenshaw56 committed:
Merge branch 'latest' into recommendation-skip-link

ryanmccombe committed:
Merge branch 'latest' into tv-cinema-mode-live

ryanmccombe committed:
Update snapshot

simonsinclair committed:
Merge branch 'latest' into secondary-column-summary-bug

rhenshaw56 committed:
Merge pull request #7302 from bbc/recommendation-skip-link

added skip link for recommendations

ryanmccombe committed:
Merge branch 'latest' into tv-cinema-mode-live

ryanmccombe committed:
Merge pull request #7206 from bbc/tv-cinema-mode-live

[DO NOT MERGE] TV Cinema Mode - Live

LilyL0u committed:
Merge branch 'latest' into adhoc-cypress-tests

rhenshaw56 committed:
removed summaary from top stories

rhenshaw56 committed:
updated top stories snapshot

rhenshaw56 committed:
updated snapshots

rhenshaw56 committed:
Merge branch 'latest' of https://github.com/bbc/simorgh into secondary-column-summary-bug

rhenshaw56 committed:
Merge branch 'secondary-column-summary-bug' of https://github.com/bbc/simorgh into secondary-column-summary-bug

Commit:
remove link tags prefetching

jimjohnsonrollings committed:
Merge branch 'latest' of github.com:bbc-news/simorgh into indexalso

rebeccamcginn committed:
Pass through path as path

Commit:
use loadable at top level for page components

Commit:
fix photogallery page tests

rebeccamcginn committed:
Merge pull request #7331 from bbc/incorrect_path_schedules

Pass path as path, not pathname

jimjohnsonrollings committed:
separate new tests from old

Commit:
Merge branch 'latest' into indexalso

jimjohnsonrollings committed:
Move data extraction into separate methods

paruchurisilpa committed:
Merge branch 'latest' into secondary-column-summary-bug

jimjohnsonrollings committed:
Split proptypes for link types

andrewscfc commented:
‘After’ is the bundle size for each page type doing a production build (`npm run build`) and start (`npm start`) from this PR: https://github.com/bbc/simorgh/pull/7311 at commit: https://github.com/bbc/simorgh/pull/7311/commits/ffda770adc72730579a4e2a2d5a90a6ae0f4a51d

‘Before’ is latest at commit: `9f13322d7ed5532ed333bae243a1a0b6fcf1a608`

Page Type | Before | After
-- | -- | --
Story http://localhost:7080/mundo/23263889 | 845.8 | 796.01
PGL http://localhost:7080/mundo/deportes-36935058 | 845.8 | 756.77
MAP http://localhost:7080/mundo/media-23283126 | 845.8 | 761.95
Article http://localhost:7080/mundo/articles/ce42wzqr2mko | 845.8 | 745.54
FrontPage http://localhost:7080/mundo | 845.8 | 772.66
MostRead http://localhost:7080/mundo/popular/read | 845.8 | 693.01
LiveRadio http://localhost:7080/nepali/bbc_nepali_radio/liveradio | 841.24 | 690.81
On Demand Radio http://localhost:7080/nepali/bbc_nepali_radio/w172x83pnptp1s8 | 841.24 | 692.61

These numbers were arrived at by manually adding up the js asset sizes in the developer tools window, the example below is for Stories with this PR:
![image](https://user-images.githubusercontent.com/9645462/88559789-32a0fc80-d025-11ea-912c-b641dcd8849b.png)

jimjohnsonrollings committed:
Merge branch 'indexalso' of github.com:bbc-news/simorgh into indexalso

jimjohnsonrollings committed:
Merge branch 'latest' of github.com:bbc-news/simorgh into 180icons

jimjohnsonrollings committed:
Add additional six pngs

andrewscfc commented:
> I guess what I am asking is will the common bundle also include components that are not common?

Hey @hindsc52 , afaik, anything that isn't common will be split out into a specific page bundle, that is on the far right of this bundle report capture - in this case it pretty much just has the `MediaAssetPage` component itself in it. The other two common bundles will contain components common to the page types listed in the name.
![image](https://user-images.githubusercontent.com/9645462/88562628-d344eb80-d028-11ea-8865-3802ec23c461.png)

jimjohnsonrollings committed:
Update manifest files

jimjohnsonrollings committed:
Update test

karinathomasbbc committed:
Merge branch 'latest' into adhoc-cypress-tests

jimjohnsonrollings committed:
Update snapshots

karinathomasbbc committed:
Improve logging in fetch page data

karinathomasbbc committed:
Move statusCodes to common location for reusability

karinathomasbbc committed:
Capture metrics on server when a non-200 response is returned from getInitialData

karinathomasbbc committed:
Merge pull request #7223 from bbc/adhoc-cypress-tests

Adhoc Cypress test project

karinathomasbbc committed:
Merge branch 'latest' into adhoc-legacy-media-playback-tests

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into toggles-file-per-environment

karinathomasbbc committed:
CPS Recommendations on Test

karinathomasbbc committed:
Cinema Mode TV on Live

karinathomasbbc committed:
Updating snapshots

jimjohnsonrollings committed:
Merge branch 'latest' into 180icons

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into adhoc-legacy-media-playback-tests

karinathomasbbc committed:
Revert "doesnt test for image if not present. Added wait to help flakiness"

This reverts commit 1216782392c421229c1ae2749c2081b54af66bce.

karinathomasbbc committed:
Merge branch 'adhoc-legacy-media-playback-tests' of github.com:bbc/simorgh into adhoc-legacy-media-playback-tests

karinathomasbbc committed:
Merge branch 'latest' into capture-cloudwatch-metric-on-404

jimjohnsonrollings committed:
Updates more snapshots

jimjohnsonrollings committed:
Temporary allow the uplaod of PNGs

ryanmccombe committed:
Update Radio availability handling to use new ARES implementation

rhenshaw56 committed:
Merge branch 'latest' into secondary-column-summary-bug

Commit:
Merge branch 'latest' into update-code-owners

Commit:
Merge pull request #7309 from bbc/update-code-owners

Removing Benjamin as code owner

rhenshaw56 committed:
Merge branch 'latest' into secondary-column-summary-bug

ryanmccombe committed:
Update fixtures

ryanmccombe committed:
Update Radio and TV page props

rhenshaw56 committed:
Merge pull request #7320 from bbc/secondary-column-summary-bug

prevent summary rendering in secondary column at medium gel widths

ryanmccombe committed:
Update cypress isExpired helper

ryanmccombe committed:
Merge branch 'latest' into od-radio-availability-refactor

ryanmccombe committed:
Update cypress helper

ryanmccombe committed:
Merge branch 'od-radio-availability-refactor' of github.com:bbc/simorgh into od-radio-availability-refactor

paruchurisilpa committed:
Merge branch 'latest' into indexalso

andrewscfc committed:
Introduce pageTypeBundleExtractor

karinathomasbbc committed:
onEnvironment can now accept an array of values, instead of just a single string

Commit:
Merge branch 'latest' into capture-cloudwatch-metric-on-404

karinathomasbbc committed:
Update test description

rebeccamcginn committed:
Removing schedules for Gahuza's Live Radio page

rebeccamcginn committed:
Merge pull request #7343 from bbc/remove_gahuza_live_radio_schedule

Removing schedules for Gahuza's Live Radio page

andrewscfc committed:
Merge branch 'latest' into indexalso

rhenshaw56 committed:
used lang to override html lang attribute in sty pages

karinathomasbbc committed:
Merge branch 'latest' into allow-multiple-environments

karinathomasbbc committed:
Merge branch 'latest' into capture-cloudwatch-metric-on-404

rhenshaw56 committed:
fix EOL

rhenshaw56 committed:
Merge branch 'latest' of https://github.com/bbc/simorgh into html-lang-overrides

RichardPK committed:
Merge branch 'latest' into od-radio-availability-refactor

andrewscfc committed:
Merge pull request #7317 from bbc/indexalso

Make index alsos which are promo assets work

LilyL0u committed:
test e2es for russian map with redirect cps url and tc2 url

karinathomasbbc committed:
Merge branch 'latest' into capture-cloudwatch-metric-on-404

LilyL0u committed:
Merge branch 'latest' into russian-map-e2es-test

karinathomasbbc committed:
Fix linting

RichardPK committed:
Merge branch 'latest' into od-radio-availability-refactor

karinathomasbbc committed:
Adding a Remote Toggle enabled column table, and links to the default values

andrewscfc committed:
working page specific bundle reporting

karinathomasbbc committed:
Merge branch 'latest' into toggles-file-per-environment

karinathomasbbc committed:
Trigger builds

RichardPK committed:
Merge pull request #7347 from bbc/russian-map-e2es-test

Test E2Es for Russian MAPs with redirect

karinathomasbbc committed:
Merge branch 'latest' into capture-cloudwatch-metric-on-404

Commit:
Merge branch 'latest' into toggles-file-per-environment

karinathomasbbc committed:
Merge pull request #7337 from bbc/capture-cloudwatch-metric-on-404

Send custom cloudwatch metrics on AReS 404 response

karinathomasbbc committed:
Merge branch 'latest' into toggles-file-per-environment

karinathomasbbc committed:
Merge branch 'latest' into adhoc-legacy-media-playback-tests

karinathomasbbc committed:
Tidy up config file - remove URLs which don't work

karinathomasbbc committed:
Merge branch 'adhoc-legacy-media-playback-tests' of github.com:bbc/simorgh into adhoc-legacy-media-playback-tests

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into toggles-file-per-environment

karinathomasbbc committed:
Merge branch 'toggles-file-per-environment' of github.com:bbc/simorgh into toggles-file-per-environment

karinathomasbbc committed:
Merge pull request #7222 from bbc/adhoc-legacy-media-playback-tests

Skeleton adhoc tests for legacy MAP media playback

karinathomasbbc committed:
Merge branch 'latest' into toggles-file-per-environment

karinathomasbbc committed:
_environment can be a string instead of an object

karinathomasbbc committed:
Updating snapshot

karinathomasbbc committed:
Merge branch 'toggles-file-per-environment' of github.com:bbc/simorgh into toggles-file-per-environment

andrewscfc committed:
Move bundlesize scripts in directory

andrewscfc committed:
Extract getPageBundleData logic to its own module

simonsinclair committed:
Merge branch 'latest' into html-lang-overrides

ryanmccombe committed:
CR feedback

ryanmccombe committed:
Merge branch 'latest' into od-radio-availability-refactor

ryanmccombe committed:
Merge pull request #7339 from bbc/od-radio-availability-refactor

Update Radio availability handling to use new ARES implementation

karinathomasbbc committed:
Remove all but preroll from the "remote toggle enabled" column

paruchurisilpa committed:
Merge branch 'latest' into html-lang-overrides

karinathomasbbc committed:
Merge branch 'latest' into toggles-file-per-environment

rhenshaw56 committed:
Merge pull request #7345 from bbc/html-lang-overrides

override html 'lang' attribute with computed 'lang' override in service config

karinathomasbbc committed:
Merge branch 'latest' into toggles-file-per-environment

rebeccamcginn committed:
Reverting OD Radio padding change

karinathomasbbc committed:
Merge pull request #7312 from bbc/toggles-file-per-environment

Separate toggles into 1 file per environment

Commit:
Merge branch 'code-split-page-type-bundles_andrew' of github.com:bbc/simorgh into code-split-page-type-bundles_andrew

RichardPK committed:
Merge branch 'latest' into allow-multiple-environments

LilyL0u committed:
live russian map urls tc2 and a cps with redirected /av/

rebeccamcginn committed:
Delete onDemand components out of RadioBlocks dir

rebeccamcginn committed:
Adding separate styling for ODTV and ODRadio

Commit:
consolidate page/main/vendor/common bundle tables

rebeccamcginn committed:
Merge branch 'latest' into odradio_heading_incorrect_padding

rebeccamcginn committed:
Merge branch 'latest' of github.com:bbc/simorgh into odradio_heading_incorrect_padding

rebeccamcginn committed:
Updating snapshots

rebeccamcginn committed:
Merge branch 'odradio_heading_incorrect_padding' of github.com:bbc/simorgh into odradio_heading_incorrect_padding

dependabot-preview[bot] committed:
[Security] Bump elliptic from 6.5.2 to 6.5.3

Bumps [elliptic](https://github.com/indutny/elliptic) from 6.5.2 to 6.5.3. **This update includes a security fix.**
- [Release notes](https://github.com/indutny/elliptic/releases)
- [Commits](https://github.com/indutny/elliptic/compare/v6.5.2...v6.5.3)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

jimjohnsonrollings committed:
Replace Japanese icon

jimjohnsonrollings committed:
Merge branch 'latest' of github.com:bbc-news/simorgh into 180icons

Commit:
Merge pull request #7342 from bbc/allow-multiple-environments

onEnvironment can now accept an array of values

rebeccamcginn committed:
Updating imports

rebeccamcginn committed:
Merge branch 'latest' into odradio_heading_incorrect_padding

FK78 committed:
Merge branch 'latest' into dependabot/npm_and_yarn/elliptic-6.5.3

jimjohnsonrollings committed:
Merge branch 'latest' into 180icons

rebeccamcginn committed:
Remove media queries, update indonesian test data

rebeccamcginn committed:
Merge branch 'odradio_heading_incorrect_padding' of github.com:bbc/simorgh into odradio_heading_incorrect_padding

simonsinclair committed:
Bump @bbc/psammead-test-helpers

simonsinclair committed:
Use jest-styled-components

LilyL0u committed:
Merge branch 'latest' into russian-map-e2es-live

FK78 committed:
Merge pull request #7357 from bbc/dependabot/npm_and_yarn/elliptic-6.5.3

[Security] Bump elliptic from 6.5.2 to 6.5.3

Commit:
Disable remote feature toggles on live

Looks like fetching remote feature toggles was accidentally enabled on live

Commit:
Merge branch 'latest' into oops

LilyL0u committed:
Merge branch 'latest' into russian-map-e2es-live

simonsinclair committed:
Prune globally available @testing-library/jest-dom/extend-expect from tests

andrewscfc committed:
Max service + page summary and re-arrange tables

Commit:
update snapshots

Commit:
Update toggles snapshot

Commit:
Merge branch 'oops' of github.com:bbc/simorgh into oops

RichardPK committed:
Merge branch 'latest' into odradio_heading_incorrect_padding

Commit:
Merge pull request #7361 from bbc/oops

Disable remote feature toggles on live

LilyL0u committed:
Merge branch 'latest' into russian-map-e2es-live

simonsinclair committed:
Resolve problematic PGL tests

LilyL0u committed:
Merge pull request #7356 from bbc/russian-map-e2es-live

[DO NOT MERGE] Live Russian MAP E2Es (a URL to redirect and a TC2)

simonsinclair committed:
Update STY tests to use established Page render pattern

jimjohnsonrollings committed:
Merge branch 'latest' into 180icons

simonsinclair committed:
Commit missing styles to snapshots

rebeccamcginn committed:
Merge branch 'latest' into odradio_heading_incorrect_padding

simonsinclair committed:
Merge branch 'latest' into test-helpers

Commit:
Enable CW metrics on live

rebeccamcginn committed:
Merge pull request #7355 from bbc/odradio_heading_incorrect_padding

Separate out padding for OD TV and Radio Heading

Commit:
Merge branch 'latest' into enable-cw-metrics-on-live

Commit:
Merge pull request #7368 from bbc/enable-cw-metrics-on-live

Enable CW metrics on live

simonsinclair committed:
Merge branch 'latest' of https://github.com/bbc/simorgh into test-helpers

simonsinclair committed:
Update snapshots following conflicts

jimjohnsonrollings committed:
Merge branch 'latest' into 180icons

simonsinclair committed:
Merge pull request #7365 from bbc/test-helpers

Jest environment improvements

jimjohnsonrollings committed:
Merge branch 'latest' into 180icons

jimjohnsonrollings committed:
Merge pull request #7315 from bbc/180icons

Add new 180px icons

Commit:
Merge branch 'latest' into code-split-page-type-bundles_andrew

Commit:
remove verbose naming from chunks

Commit:
fail CI if service + page bundles are too small/large

Commit:
fix bundleSize tests
